cmd/gpu-mutating-webhook/main.go

```diff
@@ -0,0 +1,162 @@
+/**
+# Copyright 2025 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	tlsDir      = `/etc/webhook/tls`
+	tlsCertFile = `tls.crt`
+	tlsKeyFile  = `tls.key`
+)
+
+var (
+	podResource     = metav1.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
+	gpuClaimName    = "nvidia-gpu-resourceclaim"
+	gpuTemplateName = "nvidia-gpu-resourceclaim-template"
+)
+
+func applyGPUMutation(req *admissionv1.AdmissionRequest) ([]patchOperation, error) {
+	// Only mutate if the incoming resource is a Pod CREATE request.
+	if req.Resource != podResource {
+		log.Printf("applyGPUMutation invoked for a non-Pod resource: %v", req.Resource)
```

Comment:
We need to switch to `klog` (or whatever logger we use in the rest of the code) for the final version.